V1 specification

yamahito opened the pull request:
Updating Respec and fixing errors in V1 of the spec

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -152,16 +152,16 @@
     </section>
     <section id='abstract'>
-      Whilst XQuery [[!XQUERY]] was originally envisaged and designed as a query language for XML,
+      Whilst XQuery was originally envisaged and designed as a query language for XML,
```
I think this should be: `Whilst XQuery (see [[XQuery]]) was`

yamahito replied:
This was causing validation issues as specified, possibly because the abstract/introduction counts as a normative section (see https://respec.org/docs/#normative-vs-informative-references); `[[?XQuery]]` seems to work instead?

adamretter replied:
I did test this with `[[XQUERY]]` and it does fix the validation issue.

yamahito replied:
But `[[XQUERY]]` shows up in the normative references section; is that what you want?  Otherwise `[[?XQUERY]]` populates the non-normative reference section.

Ditto for the other three references.  Let me know which (if any) you feel ought to be normative, otherwise I will leave them as non-normative. @adamretter

adamretter replied:
It seems to me that XQuery is a dependency of RESTXQ, which would make XQuery a normative reference. Can you explain why you believe it is a non-normative dependency please @yamahito ?

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -152,16 +152,16 @@
+      Whilst XQuery was originally envisaged and designed as a query language for XML,
       it has been adopted by many as a language for application development. This specification
-      describes a set of XQuery Annotations [[!XQUERY-30]] and a small set of functions to enable XQuery
+      describes a set of XQuery Annotations and a small set of functions to enable XQuery
```
I think this should be `of XQuery Annotations (see [[XQUERY-30]]) and`

adamretter replied:
I think also two more sections need to be changed from normative to informative references:
1. `namespaces (see [[!XML-NAMES]]) and referenced` (line ~234) -> `namespaces (see [[XML-NAMES]]) and referenced`.
2. `RESTXQ is heavily influenced by [[!JAX-RS]]` (line ~257) -> `RESTXQ is heavily influenced by JAX-RS (see [[JAX-RS]])`.

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -479,8 +479,8 @@ <h4>Resource Function Parameters</h4>
               <a href="#form-param-annotation">Form</a> or
-              <a href="#header-param-annotation">Header</a> parameters, or zero or one
-              atomic value in the case of <a href="#cookie-param-annotation">Cookie</a>
+              <a href="#form-header-annotation">Header</a> parameters, or zero or one
```
There is no such thing as a "Form Header Annotation" in RESTXQ. This was correct and should indeed be `header-param-annotation` - it is the other end of the link that needs to be fixed please.

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -479,8 +479,8 @@ <h4>Resource Function Parameters</h4>
-              <a href="#header-param-annotation">Header</a> parameters, or zero or one
-              atomic value in the case of <a href="#cookie-param-annotation">Cookie</a>
+              <a href="#form-header-annotation">Header</a> parameters, or zero or one
+              atomic value in the case of <a href="#form-cookie-annotation">Cookie</a>
```
There is no such thing as a "Form Cookie Annotation" in RESTXQ. This was correct and should indeed be `cookie-param-annotation` - it is the other end of the link that needs to be fixed please.

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -886,6 +886,8 @@ <h4>rest:uri()</h4>
       </section>
     </section>
+    
+    <section id="conformance"/>
```
I think this is missing some additional conformance information... How about something like:

```html
    <section id="conformance">
      <p>This section defines the conformance criteria for a RESTXQ 1.0 implementation. An implementation that claims to conform to this specification MUST include a claim of Minimal Conformance as defined in <a href="#minimal-conformance" class="sectionRef"></a>.</p>


      <section id="minimal-conformance">
        <h3>Minimal Conformance</h3>
        <p>An implementation that claims <strong>Minimal Conformance</strong> to this specification MUST provide all of the following items:</p>
        <ol>
          <li><p>An implementation of everything specified in this document.</p></li>
          <li><p>A definition of every item specified to be implementation-defined.</p></li>
        </ol>
      </section>
    </section>
```

yamahito replied:
Happy to add this; just wanted to note that the empty section is automatically expanded by respec with the standard text:

> As well as sections marked as non-normative, all authoring guidelines, diagrams, examples, and notes in this specification are non-normative. Everything else in this specification is normative.
>
> The key words MAY, MUST, MUST NOT, and SHOULD in this document are to be interpreted as described in [BCP 14](https://datatracker.ietf.org/doc/html/bcp14) [[RFC2119](file:///Users/yamahito/Projects/exquery/exquery-restxq-specification/restxq-1.0-specification.html#bib-rfc2119)] [[RFC8174](file:///Users/yamahito/Projects/exquery/exquery-restxq-specification/restxq-1.0-specification.html#bib-rfc8174)] when, and only when, they appear in all capitals, as shown here.

(which will be pre-pended to the result)

adamretter replied:
Yes, I did test it, please add it.

yamahito replied:
Already added below: https://github.com/exquery/exquery/pull/151/commits/1ebc73cb5c322cd715887d924f08d4b066597857

adamretter reviewed (changes requested):
Thanks. See feedback comments for some corrections that are needed please.

adamretter commented:
@yamahito Can you update your commit messages to the standard form we use please (see: https://github.com/eXist-db/exist/blob/develop/CONTRIBUTING.md#commit-labels)

adamretter commented:
> I have rebased to reword the commit messages, but it does still leave the old commits in the PR history

@yamahito Erm I am not really sure how that is possible... Can you describe what you did exactly and maybe I can help figure out what went wrong here?

yamahito commented:
> > I have rebased to reword the commit messages, but it does still leave the old commits in the PR history
> 
> @yamahito Erm I am not really sure how that is possible... Can you describe what you did exactly and maybe I can help figure out what went wrong here?
> 
> p.s. Your commit messages seems to still have a `:` after the label, and they should consistently start with a capital letter after the label please.

Never mind, fixed it!  I will go and amend those colons and capitals now I know how to do what I need.

yamahito commented:
@adamretter I've been able to rebase my local git and force push the reworded commit messages to this branch: I believe that all the changes with the correct commit history (and messages) are now committed.  I will add those changes from K-2963 to this PR that you've identified elsewhere and re-request the review.

yamahito commented:
@adamretter I have now added the actions from K-2963 and this is again ready for review.

There was some refactoring of the `%rest:consumes` and `%rest:produces` annotations to group them with other annotation definitions, which is why I think the content negotiation preamble was missing.  I have elected to add this as a new section, which could be expanded upon in the 2.0 spec.  Let me know if you feel this is the wrong approach.

I also spotted what I believe are some spelling errors, again, please let me know if this is specialised vocabulary that I have mistakenly "corrected".

adamretter commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -713,6 +723,23 @@ <h3>Base URI</h3>
+              fn:collection("/db/widgets")/widgets
+            };
+          </pre>
+        <p>The mechanism by which formats are converted are implementation dependent.</p>
```
I don't think that is entirely correct, as it is also influenced by the `%output:method` annotation. I would suggest removing this paragraph please.

yamahito replied:
OK; I wanted to clarify that the (automatic) representation of e.g. an XML fragment as a JSON map (such as is supported natively by eXist-db) was outside the scope of the specification.  For now I will remove it as you suggest.

adamretter reviewed (changes requested):
Good improvements, but some changes still needed please.

yamahito commented on exquery-restxq-specification/restxq-1.0-specification.html:
```diff
@@ -903,6 +903,13 @@ <h4>rest:uri()</h4>
           </p>
         </section>
+        <section>
+          <h4>rest:build-uri()</h4>
```
whoops.  Will fix.

adamretter reviewed (approved):
LGTM. Thanks @yamahito
